datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,268 @@
+                if (avroTypeName.equalsIgnoreCase("ARRAY")) {
+                  if (avroTypeSchema.getElementType().getType().getName().equalsIgnoreCase("RECORD")) {
+                    // get schema of nested obj and generic record
+                    List<GenericRecord> nestedObjectArray = new ArrayList<GenericRecord>();
+                    // In order to overcome warning: [unchecked] unchecked cast, suppressing it as we are sure that it will be array<record>
+                    @SuppressWarnings("unchecked")
+                    GenericArray<GenericRecord> array = (GenericArray<GenericRecord>) avroRecord.get(fieldName);
+                    if (array != null) {
+                      ListIterator<GenericRecord> it = array.listIterator();
+                      while (it.hasNext()) {
+                        GenericRecord nestedObjectRecord = it.next();
+                        GenericRecord parquetFormatRecord = generateParquetStructuredAvroData(avroTypeSchema.getElementType(), nestedObjectRecord);
+                        nestedObjectArray.add(parquetFormatRecord);
+                      }
+                    }
+                    record.put(fieldName, nestedObjectArray);
+                  } else {
+                    record.put(fieldName, avroRecord.get(fieldName));
+                  }
+                } else if (avroTypeName.equalsIgnoreCase("LONG")) {
+                  // For timestamp fields we need to check for logical type in avroschema, to convert to string in parquet
+                  // for testing with 1.0 events hardcoded eventTimestamp in order to make it human readable
+                  if (field.name().equalsIgnoreCase("eventTimestamp") ||
+                      (avroTypeSchema.getLogicalType() != null &&
+                          avroTypeSchema.getLogicalType().getName().equalsIgnoreCase("timestamp-millis"))) {
+                    // For now converting event timestamp to datetimeFormat
+                    // but in 2.0 we still need to figure out how we are handling timestamps
+                    String result = null;
+                    if (avroRecord.get(fieldName) != null) {
+                      Long value = (Long) avroRecord.get(fieldName);
+                      result = convertDateTimeToString(value);
+                    }
+                    record.put(fieldName, result);
+                  } else {
+                    // this is for primitive data types which do not need any conversion
+                    record.put(fieldName, avroRecord.get(fieldName));
+                  }
+                  // Using ProductManufacturerPart to test for 1.0 event, will replace with header name for 2.0
+                } else if (avroTypeName.equalsIgnoreCase("RECORD") && avroTypeSchema.getName().equalsIgnoreCase("ProductManufacturerPart")) {
+                  GenericRecord nestedRecord = (GenericRecord) avroRecord.get("manufacturerPart");
+                  record.put(fieldName, nestedRecord.get(fieldName));
```

Comment:
replace `ProductManufacturerPart` with scribeHeader if we are splitting header object

Reply:
replaced with scribeHeader